### DISCUSSION TITLE
POP消费的性能

### DISCUSSION THREAD

**Discussion by milksteven** [apache/rocketmq#6360] [Q&A] [answered]
我在做性能压测的时候，发现同一个Topic，使用不同的消费组进行消费，POP消费的TPS只有Pull的一半，这种是否正常？

**Comment by milksteven** [comment]
我目前想到的影响的点有：
1. 共享队列，涉及到加解锁
2. 服务端ACK，网络延迟增加了耗时链路

社区有POP和Pull的性能压测结果分享？

**Comment by RongtongJin** [comment, marked as answer]
压测是否开启了enablePopBufferMerge ，开启该参数可以使得broker中Buffer Merge的策略生效，ck和ack尽量在内存中抵消。
理论上，相同consumer数量下，POP性能可能要略低于PULL，POP消费链路更加复杂，内存中无法merge还会走落盘的消息链路。Pull只是单纯的拉队列中的消息而已。但POP负载均衡是消息粒度而非队列粒度，POP消费其实可以无限扩展消费端增加消费能力。
